경북대 FE_김민주_2주차 과제 Step 1

joojjang opened the pull request:
구현한 부분은 다음과 같습니다. 
- [x] Header, Footer 컴포넌트 구현
- [ ] 라우팅
  메인 페이지, Theme 페이지, 로그인 페이지, 나의 페이지
- [ ]

joojjang committed:
feat(components): Header 구현

- Header 컴포넌트 추가
- Emotion으로 스타일링
- 추가적인 전역 css 속성이 필요하여 reset.css 수정

joojjang committed:
feat(components): Footer 구현

- Footer 컴포넌트 구현
- Emotion으로 스타일링
- App.tsx에 임포트

joojjang committed:
feat(router): react-router-dom 설치, 라우트 연결

- App.tsx에 home, theme, login, my-account 페이지를 라우팅
- 각 페이지는 아직 만들지 않은 상태

joojjang committed:
feat(layout): 전체 화면 레이아웃 구현

- Header, Footer 들어 있는 전체  화면 레이아웃
- 루트 경로에 라우팅

joojjang committed:
feat(pages): Home 페이지 구현

- Container 재사용하기 위해 코드 수정
- 배너 2개만 구현, theme과 선물랭킹은 아직

joojjang committed:
style(eslint): 'simple-import-sort/imports' 룰 warn으로 변경

- error -> warn
- 컴포넌트 구현 중에 원활한 테스트 위함

joojjang committed:
feat(pages): Home 페이지 내 theme 섹션 구현

danmin20 commented:
step1 PR만 올리신 걸까요?

danmin20 commented:
README 파일 + 퀴즈에 대한 답변이 없는 것 같아요~

danmin20 commented on src/components/common/layouts/Container/index.tsx:
```diff
@@ -5,13 +5,25 @@ import { vars } from '@/styles';
+  flexDirection?: 'row' | 'column';
+  justifyContent?: 'center' | 'flex-start' | 'flex-end' | 'space-between' | 'space-around';
+  alignItems?: 'center' | 'flex-start' | 'flex-end' | 'baseline' | 'stretch';
 } & React.HTMLAttributes<HTMLDivElement>;
```
& React.HTMLAttributes<HTMLDivElement> 를 추가해주신 이유가 있을까요?

danmin20 commented on src/components/features/Home/ThemeCategorySection/index.tsx:
```diff
@@ -0,0 +1,130 @@
+      <div>
+        <Container>
+          <Grid columns={4}>
+            <ThemeCategoryItem>
+              <CategoryImage
+                src="https://img1.daumcdn.net/thumb/S104x104/?fname=https%3A%2F%2Ft1.daumcdn.net%2Fgift%2Fhome%2Ftheme%2F292020231106_MXMUB.png"
+                alt="생일"
+              />
+              <CategoryTitle>생일</CategoryTitle>
+            </ThemeCategoryItem>
```
반복 코드가 많은 것 같아요~ map을 사용해서 효율적으로 작성할 수 있을 것으로 보여요.

joojjang commented:
엇 구글폼만 내놓고 아직 구현 중이었습니다..

joojjang committed:
feat(pages): Home 페이지 내 랭킹 섹션 구현

- 랭킹 필터 onClick 함수 미구현
- 더보기 버튼 구현
- 중복 코드 모듈화 필요

joojjang committed:
chore(components): GoodsItems 컴포넌트명과 파일명 일치화

joojjang committed:
feat(pages): Login 페이지 구현

joojjang committed:
fix(routes): App.tsx에서 라우트 수정

- Login 페이지에서 Header와 Footer가 담긴 Layout이 안 나오도록 Layout 외부로 뺌
- RouterPath 변수를 생성, 없는 경로 추가

joojjang committed:
feat(pages): MyAccount 페이지 구현

- 로그아웃 버튼 클릭 시 홈으로 리다이렉트 구현(페이지 이동만)
- 사용자 이름 실제로 들어가도록 구현 필요
